### PR TITLE
feat(client): event subprocess support

### LINKED ISSUE #86: I can model an event sub process
I can add an event sub process to the diagram. The event sub process can be triggered by the following events:

- Interrupting/Non-Interrupting Message Event (same configuration as intermediate message catch event)
- Interrupting/Non-Interrupting Timer Event (same configuration as intermediate timer event)
- Interrupting Error Event (same configuration as error boundary event #85)

__Describe the solution you'd like__

* [ ] Add event sub process as replace option for normal sub process, Hint: Have a look at the original [`ReplaceMenuProvider`](https://github.com/bpmn-io/bpmn-js/blob/develop/lib/features/popup-menu/ReplaceMenuProvider.js) and the zeebe related [`Options`](https://github.com/zeebe-io/zeebe-modeler/blob/master/client/src/app/tabs/bpmn/custom/modeler-options/Options.js)
* [ ] Add rules for available start events inside an event sub process, Hint: Have a look at how we [distinguish the available options](https://github.com/bpmn-io/bpmn-js/blob/develop/lib/features/popup-menu/ReplaceMenuProvider.js#L99) for start events inside and outside event sub processes, and decide how to modify the [available options](https://github.com/zeebe-io/zeebe-modeler/blob/master/client/src/app/tabs/bpmn/custom/modeler-options/Options.js) in order to only allow the given possibilities (see above)

### PR DESCRIPTION
**Pull request by oguzeroglu**
Closes #86

This PR adds following elements into Zeebe modeller:

- Event subprocess

- Error start event

- Non-interrupting message start event

- Non interrupting timer start event

I also updated Error event related test cases by adding test cases for error start event.

Kapture 2019-09-06 at 15 43 56

### CODE REVIEW

File: client/src/app/tabs/bpmn/custom/CustomReplaceMenuProvider.js [hunk 1]
```diff
@@ -40,8 +36,7 @@ export default class CustomReplaceMenuProvider extends ReplaceMenuProvider {
     let headerEntries = [];
 
     if (
-      isAny(element, [ 'bpmn:ReceiveTask', 'bpmn:ServiceTask', 'bpmn:SubProcess' ]) &&
-      !isEventSubProcess(element)
```
Comment (pinussilvestrus) [inline client/src/app/tabs/bpmn/custom/CustomReplaceMenuProvider.js]: Do we want to have the possibility to add `multiInstance` marker to EventSubProcesses? It's not possible in the Camunda Modeler.

Comment (oguzeroglu) [reply client/src/app/tabs/bpmn/custom/CustomReplaceMenuProvider.js]: Makes sense, bringing that condition back in now

Comment (oguzeroglu) [reply client/src/app/tabs/bpmn/custom/CustomReplaceMenuProvider.js]: @pinussilvestrus Done!

Comment (Zelldon) [reply client/src/app/tabs/bpmn/custom/CustomReplaceMenuProvider.js]: Hey guys, I think it doesn't make sense to have multi instance marker on an event subprocess.

Comment (pinussilvestrus) [reply client/src/app/tabs/bpmn/custom/CustomReplaceMenuProvider.js]: yep, that's why we decided to not remove the condition 👍

Comment (Zelldon) [reply client/src/app/tabs/bpmn/custom/CustomReplaceMenuProvider.js]: Ah ok sorry then I misunderstood the comments sorry :slightly_smiling_face:

Comment (oguzeroglu) [reply client/src/app/tabs/bpmn/custom/CustomReplaceMenuProvider.js]: Yep, this is the current behaviour now:

Kapture 2019-09-09 at 11 51 28

**Comment by pinussilvestrus** [comment]
One additional thing: we should modify the [`CustomContextPadProvider`](https://github.com/zeebe-io/zeebe-modeler/blob/develop/client/src/app/tabs/bpmn/custom/CustomContextPadProvider.js) to _not_ offer any appending-options for an EventSubProcess

Zeebe Modeler (current)

image

Camunda Modeler

image

**Comment by nikku** [comment]
Yes, we should. Event-Sub-Processes cannot be connected.

**Comment by oguzeroglu** [comment]
@pinussilvestrus Done! Also added following test cases for Event SubProcess:

1) Cannot have append options in pad entries
2) Can have only Sub Process in replace options and cannot have Sequential/MultiInstance entries
3) A BoundaryEvent cannot be attached to an Event Subprocess

**Pull request merged**